### DISCUSSION TITLE
What prevents Stream Chat Response from working?

### DISCUSSION THREAD

**Discussion by vulcan4d** [open-webui/open-webui#7707] [General]
I have an Ubuntu VM under ProxMox with dedicated GPUs
I then have a Docker container with OpenWebUI
There is no proxy, all internal and on the same subnet 
Why won't it stream responses when enabled?

The responses here say it's due to NGINX and a proxy though I'm not using one.  It is pretty basic stuff but I read that stream response is the default feature that is just expected to work.  Any ideas would be greatly appreciated.

**Comment by smonux** [comment]
It's hard to give advice because there is little information. I would start debugging in the web console, and see if stream=true is being set when forced.

If that's the case, I would start looking at the OpenWebui->backend connection. This is the trick I do for this, run mitmproxy in a port like this

mitmweb --listen-port 8082 --web-port 9090  --mode reverse:http://localhost:11434

Now, disable the old connection and point to the proxied one:
![imagen](https://github.com/user-attachments/assets/915aed4f-be8b-4438-9383-1b00a38f0782)

You can now connect to the interface and inspect the request and the response:
![imagen](https://github.com/user-attachments/assets/a9e7e1c5-1ee7-43de-a018-5b3609b028a3)

![imagen](https://github.com/user-attachments/assets/561affde-ee30-46df-8538-8402b298f22f)

You may have to change the params a bit for your case,  the mitmweb by default it's only available on localhost and when it's in a server it's not directly accessible from other hosts.  I do a tunnel with ssh -L, but there are other options.

That should help you identify the failing component, at least.

Good luck
